### PR TITLE
Propogate failures in weather requests

### PR DESCRIPTION
**Pull request by janua**
This removes the `.fail` methods on `weather` requests in `weather.js`.

`msg` is always `undefined` so these `fails` are giving no useful information in `sentry`.

This should allow them to bubble up to `raven` elsewhere to get properly reported to `sentry`.

@OliverJAsh

### CODE REVIEW

**Comment by OliverJAsh** [comment]
Does it matter that this halts JS execution?

**Comment by janua** [comment]
Does it? I thought these `Promises` would keep it wrapped?

File: static/src/javascripts/projects/facia/modules/onwards/weather.js [hunk 1]
```diff
@@ -136,16 +123,7 @@ define([
             return this.getWeatherData(config.page.forecastsapiurl + '/' + location.id + '.json?_edition=' + config.page.edition.toLowerCase())
                 .then(function (response) {
                     this.renderForecast(response);
-                }.bind(this))
-                .fail(function (err, msg) {
-                    var statusText = (err && err.statusText) || '';
-                    var statusCode = (err && err.status) || '';
```
Comment (OliverJAsh) [inline static/src/javascripts/projects/facia/modules/onwards/weather.js]: `err` might be an `Error` object with custom properties set. In which case, we'll have a `message` and `stack` prop.

File: static/src/javascripts/projects/common/utils/ajax-promise.js [hunk 2]
```diff
@@ -13,6 +15,11 @@ define([
             })
             .fail(function (request, text, err) {
                 var error = err ? err : new Error(text);
+
+                var statusText = (err && err.statusText) || '';
+                var statusCode = (err && err.status) || '';
+                raven.captureException('Error retrieving data (' + text + ') (Status: ' + statusCode + ') (StatusText: ' + statusText + ')');
```
Comment (janua) [inline static/src/javascripts/projects/common/utils/ajax-promise.js]: @OliverJAsh Is this what you had in mind?

**Comment by janua** [comment]
Now using `ajax-promise` in `weather`.

File: static/src/javascripts/projects/common/utils/ajax-promise.js (hunk 2 above)
Comment (janua) [inline static/src/javascripts/projects/common/utils/ajax-promise.js]: Forgot to mention, I have checked for usages of `fail` throughout the project off `ajax-promise` and didn't really find any, for the case of double `raven` reporting.

The only thing I don't like about this is that we loose the `weather` tag in `raven`.

Comment (OliverJAsh) [inline static/src/javascripts/projects/common/utils/ajax-promise.js]: I would make this the message, i.e. pass it into `new Error(<msg>)`.

Then, let the consumer decide whether to `raven.captureException`.

Comment (OliverJAsh) [inline static/src/javascripts/projects/common/utils/ajax-promise.js]: BTW: The `when` Promise library supports a global event handler for unhandled promise rejections, so I have started migrating us to that.

File: static/src/javascripts/projects/facia/modules/onwards/weather.js [hunk 3]
```diff
@@ -102,14 +102,7 @@ define([
...
                         omniture.trackLinkImmediate(true, 'o', 'weather location set by fastly');
-                    }.bind(this))
-                    .fail(function (err, msg) {
-                        raven.captureException(new Error('Error retrieving city data (' + msg + ')'), {
-                            tags: {
-                                feature: 'weather'
-                            }
-                        });
-                    });
+                    }.bind(this));
```
Comment (OliverJAsh) [inline static/src/javascripts/projects/facia/modules/onwards/weather.js]: Don't you want to capture the error?

``` js
.fail(() => raven.captureException…
```

Comment (janua) [inline static/src/javascripts/projects/facia/modules/onwards/weather.js]: Oops yes. I see what you meant now originally

**Comment by OliverJAsh** [comment]
Perfect :kiss:

**Comment by OliverJAsh** [comment]
:+1:

**Pull request merged**